### PR TITLE
Create NearestRadiusLayeredAverage

### LINKED ISSUE #16829: Create NearestRadiusLayeredAverage
## Reason

For cases where it is desired to have layered averages done for concentric rings, this would be quite useful.

## Design

Same design as `NearestRadiusLayeredAverage` but the distance would be computed based on radii.

## Impact

Enhances current capabilities.

### PR DESCRIPTION
**Pull request by vincentlaboure**
Closes #16829

### CODE REVIEW

File: framework/src/userobject/NearestRadiusLayeredAverage.C [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "NearestRadiusLayeredAverage.h"
+#include "LayeredAverage.h"
+
+registerMooseObject("MooseApp", NearestRadiusLayeredAverage);
+
+defineLegacyParams(NearestRadiusLayeredAverage);
```
Comment (loganharbour) [inline framework/src/userobject/NearestRadiusLayeredAverage.C:16]:
```suggestion
```

File: framework/src/userobject/NearestRadiusLayeredAverage.C [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+InputParameters
+NearestRadiusLayeredAverage::validParams()
+{
+  InputParameters params =
+      nearestPointBaseValidParams<LayeredAverage, ElementIntegralVariableUserObject>();
+
+  params.set<MooseEnum>("dist_norm") = "radius";
```
Comment (loganharbour) [inline framework/src/userobject/NearestRadiusLayeredAverage.C:24]: Should this not be suppressed too?

**Review by loganharbour — CHANGES_REQUESTED** [review]

**Comment by vincentlaboure** [comment]
Comments should be addressed. Thanks for the quick review @loganharbour!

**Review by loganharbour — APPROVED** [review]

**Comment by vincentlaboure** [comment]
Thanks @loganharbour!

**Pull request merged**